Model/Api/DiscountCodeValidation.php

```diff
@@ -251,7 +260,7 @@ public function validate()
                 // check if the cart / quote exists and it is active
                 try {
                     /** @var Quote $parentQuote */
-                    $parentQuote = $this->cartHelper->getActiveQuoteById($parentQuoteId);
+                    $parentQuote = $this->cartHelper->getQuoteById($parentQuoteId);
```

Comment:
This no longer checks the active flag right? This is a pretty big functional change - is this the right way to go about the fix?

Reply:
Added asana
https://app.asana.com/0/0/1159061667010227/f

Reply:
@matt-thomason this was s a small trade-off needed to support discounts from product page checkout. Often we can have 2 quotes, the session one (and its immutable clones) and the quote we create for ppc. Actually we can have many ppc quotes since one is created on every click on buy now button. We can eventually reduce their numbers by caching them (created asana task https://app.asana.com/0/941920570700290/1159123845531097/f). Anyway, we have to create ppc quotes as inactive, otherwise all kind of weird situations happen, especially for the logged in customer, ie. interference with the session quote. I added a fix, loading inactive quote only for ppc (immutable quote id == quote id).